File: backend/mallapi/src/main/java/org/nareun130/mallapi/controller/TodoController.java
package org.nareun130.mallapi.controller;

import org.nareun130.mallapi.dto.PageRequestDTO;
import org.nareun130.mallapi.dto.PageResponseDTO;
import org.nareun130.mallapi.dto.TodoDTO;
import org.nareun130.mallapi.service.TodoService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import lombok.RequiredArgsConstructor;
import lombok.extern.log4j.Log4j2;


@RestController
@RequiredArgsConstructor
@Log4j2
@RequestMapping("/api/todo")
public class TodoController {
    
    private final TodoService service;

    @GetMapping("/{tno}")
    public TodoDTO get(@PathVariable(name = "tno") Long tno) {
        
        return service.get(tno);
    }

    @GetMapping("/list")
    public PageResponseDTO<TodoDTO> list (PageRequestDTO pageRequestDTO) {

        log.info(pageRequestDTO);

        return service.list(pageRequestDTO);
    }
    
    
}

File: backend/mallapi-kotlin/src/main/kotlin/com/nareun/mallapikotlin/controller/TodoController.kt
package com.nareun.mallapikotlin.controller

import com.nareun.mallapikotlin.dto.PageRequestDTO
import com.nareun.mallapikotlin.dto.PageResponseDTO
import com.nareun.mallapikotlin.dto.TodoDTO
import com.nareun.mallapikotlin.service.TodoService
import lombok.extern.log4j.Log4j2
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@Log4j2
@RequestMapping("/api/todo")
class TodoController(private val service: TodoService) {

    private val log = LoggerFactory.getLogger(TodoController::class.java)
//    @Autowired
//    private lateinit var todoService: TodoService

    @GetMapping("/{tno}")
    fun get(@PathVariable("tno") tno: Long): TodoDTO = service.get(tno)

    @GetMapping("/list")
    fun list(pageRequestDTO: PageRequestDTO): PageResponseDTO<TodoDTO> {
        log.info(pageRequestDTO.toString())
        return service.list(pageRequestDTO)
    }

    @PostMapping("/")
    fun register(@RequestBody todoDTO: TodoDTO): Map<String, Long?> {
        log.info("TodoDTO : $todoDTO")

        val tno = service.register(todoDTO)
        return mapOf("TNO" to tno)
    }

    @PutMapping("/{tno}")
    fun modify(@PathVariable("tno") tno: Long, @RequestBody todoDTO: TodoDTO): Map<String, String> {
        todoDTO.tno = tno

        log.info("Modify : $todoDTO")

        service.modify(todoDTO)

        return mapOf("RESULT" to "SUCCESS")
    }

    @DeleteMapping("/{tno}")
    fun remove(@PathVariable("tno") tno: Long) : Map<String,String>{
        log.info("Remove: $tno")
        service.remove(tno)

        return mapOf("RESULT" to  "SUCCESS")
    }
}